tests/v2/validation/scaling/rke2_custom_cluster_add_nodes_test.go

```diff
@@ -0,0 +1,275 @@
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	appv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type RKE2CustomClusterAddNodeTestSuit struct {
+	suite.Suite
+	session        *session.Session
+	client         *rancher.Client
+	clustersConfig *provisioninginput.Config
+}
+
+func (c *RKE2CustomClusterAddNodeTestSuit) TearDownSuite() {
+	c.session.Cleanup()
+}
+
+func (c *RKE2CustomClusterAddNodeTestSuit) SetupSuite() {
+	testSession := session.NewSession()
+	c.session = testSession
+
+	c.clustersConfig = new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, c.clustersConfig)
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(c.T(), err)
+
+	c.client = client
+}
+
+func (c *RKE2CustomClusterAddNodeTestSuit) TestRKE2CustomClusterAddWorkerNode() {
+	subSession := c.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := c.client.WithSession(subSession)
+	require.NoError(c.T(), err)
+
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(c.T(), clusterName, "Cluster name is not provided")
+
+	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
+	require.NoError(c.T(), err)
+
+	daemonsetName := namegen.AppendRandomString("test-ds")
+
+	nodeProviderNames := c.clustersConfig.NodeProviders
+	for _, nodeProviderName := range nodeProviderNames {
+		externalNodeProvider := provisioning.ExternalNodeProviderSetup(nodeProviderName)
+		steveClient, err := client.Steve.ProxyDownstream(clusterID)
+		require.NoError(c.T(), err)
+
+		initialWorkerNodeCount, err := client.Management.Node.List(&types.ListOpts{
+			Filters: map[string]interface{}{
+				"clusterId": clusterID,
+				"worker":    true,
+			},
+		})
+		require.NoError(c.T(), err)
+
+		containerTemplate := workloads.NewContainer("rancher-test", "ranchertest/mytestcontainer", corev1.PullAlways, []corev1.VolumeMount{}, []corev1.EnvFromSource{})
+		podTemplate := workloads.NewPodTemplate([]corev1.Container{containerTemplate}, []corev1.Volume{}, []corev1.LocalObjectReference{}, nil)
+		daemonsetTemplate := workloads.NewDaemonSetTemplate(daemonsetName, workloadNS, podTemplate, true, nil)
+		daemonsetObj, err := steveClient.SteveType(workloads.DaemonsetSteveType).Create(daemonsetTemplate)
+		require.NoError(c.T(), err)
+		assert.Equal(c.T(), daemonsetObj.Name, daemonsetName)
+
+		err = charts.WatchAndWaitDaemonSets(client, clusterID, workloadNS, metav1.ListOptions{
+			FieldSelector: "metadata.name=" + daemonsetObj.Name,
+		})
+		require.NoError(c.T(), err)
+
+		c.validateDaemonset(client, clusterID, workloadNS, daemonsetObj)
+		logrus.Infof("daemonset created successfully")
```

Comment:
this is separate validation that is partially covered in a new PR that Markus has up here: https://github.com/rancher/rancher/pull/42515/files#diff-3093de31f7f91239431265f2b2c8b57c13c10550e824d46de1e05133cfb2f3a8R23-R43. Is there a reason that you're adding a new daemonset as part of this test? If not, I think we should wait for that to merge, rebase your PR and use that function as he and ToU worked on it together. If so, then this should still be pulled out into a separate function, something like `createAndValidateDaemonset`.